### PR TITLE
Add implementation for CC-TripTastic

### PR DESCRIPTION
**Pull request by DenialRiver1434**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/cc-TRPTSTIC.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		if (test(mid)) {
+			hi = mid;
+		} else {
+			lo = mid + 1;
+		}
+	}
+	return lo;
+}
+
+void solve() {
```
Comment (SansPapyrus683) [inline solutions/silver/cc-TRPTSTIC.mdx:76]: just put this in `main()`

File: solutions/silver/cc-TRPTSTIC.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			if (j > 0) { pref[i][j] += pref[i][j - 1]; }
+			if (i > 0 && j > 0) { pref[i][j] -= pref[i - 1][j - 1]; }
+		}
+	}
+
+	cout << first_true(0, max(N, M)) << endl;
+}
+
+int main() {
+	ll T;
```
Comment (SansPapyrus683) [inline solutions/silver/cc-TRPTSTIC.mdx:118]: idt this has to be an `ll`
same w/ many other variables you defined i think

File: solutions/silver/cc-TRPTSTIC.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	// Setting up the size of the prefix sum array
+	pref.clear();
+	for (ll i = 0; i < N; i++) {
+		vector<ll> row(M, 0);
+		pref.push_back(row);
+	}
+
+	// Filling in the prefix sum array
+	for (ll i = 0; i < N; i++) {
```
Comment (SansPapyrus683) [inline solutions/silver/cc-TRPTSTIC.mdx:105]: yeah like this one

File: solutions/silver/cc-TRPTSTIC.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	if (tot < (K + 1)) {
+		cout << -1 << endl;
+		return;
+	}
+
+	// Setting up the size of the prefix sum array
+	pref.clear();
+	for (ll i = 0; i < N; i++) {
+		vector<ll> row(M, 0);
+		pref.push_back(row);
```
Comment (SansPapyrus683) [inline solutions/silver/cc-TRPTSTIC.mdx:101]:
```suggestion
		pref.push_back(vector<ll>(M));
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/cc-TRPTSTIC.mdx (hunk 3 above)
Comment (DenialRiver1434) [reply solutions/silver/cc-TRPTSTIC.mdx]: yeah you're definitely right, force of habit lol

File: solutions/silver/cc-TRPTSTIC.mdx (hunk 1 above)
Comment (DenialRiver1434) [reply solutions/silver/cc-TRPTSTIC.mdx]: A lot of USACO editorials are written this way, but I have adjusted it for now

File: solutions/silver/cc-TRPTSTIC.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+int N, M, K;
```
Comment (SansPapyrus683) [inline solutions/silver/cc-TRPTSTIC.mdx:20]: make lowercase

**Review by SansPapyrus683 — APPROVED** [review]
just this one thing

**Comment by DenialRiver1434** [comment]
I have finished the changed N, M, K to lowercase, and moved the problem to the binary search module (https://github.com/cpinitiative/usaco-guide/issues/3749). 
The difficulty is updated to "Normal" as it has about the same difficulty as Multiplication Table as it requires the use of 2D Prefix Sums.

**Comment by DenialRiver1434** [comment]
@SansPapyrus683 please let me know if you would still like me to do anything.

**Review by envyaims — APPROVED** [review]
kinda cursed because i'd just stick to one indexing but ok

**Pull request merged**